### PR TITLE
[Feature] Added masking views feature in Delfin UI.

### LINKED ISSUE #593: [Delfin UI] Support Host Mapping management for Heterogeneous framework from Dashboard
**Issue/Feature Description:**
Display the Host mapping alongwith the relationship between resources and host.

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
Delfin project will be adding support for Host mapping and provide APIs for host mapping management for the Heterogeneous Storage Framework.

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
/kind new feature

**What this PR does / why we need it**:
This PR adds support for masking views in Delfin UI.
User will be able to list the masking views and in turn the resources that are masked using the masking view.

**Which issue(s) this PR fixes**:
Fixes #593 

**Test Report Added?**:
/kind TESTED

**Test Report**:
masking-views-1
masking-views-list
masking-views-ports-list
masking-views-ports-list-expand
masking-views-storage-host-details
masking-views-volumes-list
masking-views-volumes-list-expand

**Special notes for your reviewer**:

### CODE REVIEW

File: src/app/business/delfin/delfin.service.ts [hunk 1]
```diff
@@ -435,5 +435,114 @@ export class DelfinService {
     metricsConfig(id, params): Observable<any>{
         return this.http.put(this.delfinStoragesUrl + '/' + id + '/metrics-config', params);
     }
+
+    //Get All Masking Views
+    
+    getAllMaskingViews(storageId, nativeMaskingViewId?, nativeStorageHostId?, nativeStorageHostGroupId?, nativeStorageVolumeGroupId?, nativeStoragePortGroupId?, name?, limit?, offset?, sort?): Observable<any> {
```
Comment (sushanthakumar) [inline src/app/business/delfin/delfin.service.ts:441]: Currently nativeStorageVolumeGroupId, nativeStoragePortGroupId filters not supported, hope it does not affect

File: src/app/business/delfin/masking-views/masking-views.component.html [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                <ng-template pTemplate="body" let-maskingView="rowData">
+                    <a [routerLink]="" (click)="showDetailsView(maskingView.volumes, 'volumesList')"> View </a>
+                </ng-template>
+            </p-column>
+            <p-column header="Ports" [style]="{'width': '10%'}">
+                <ng-template pTemplate="body" let-maskingView="rowData">
+                    <a [routerLink]="" (click)="showDetailsView(maskingView.ports, 'portsList')"> View </a>
+                </ng-template>
+            </p-column>
+            <p-column header="Initiators" [style]="{'width': '10%'}">
```
Comment (sushanthakumar) [inline src/app/business/delfin/masking-views/masking-views.component.html:62]: "Storage Host Initiators"

File: src/app/business/delfin/masking-views/masking-views.component.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        name: "Name",
+        description: "Description",
+        createdAt: "Created At",
+        updatedAt: "Updated At",
+        hostName: "Name",
+        ip: "IP Address",
+        port: "Port",
+        status: "Status",
+        osType: "OS",
+        accessMode: "Access Mode",
```
Comment (sushanthakumar) [inline src/app/business/delfin/masking-views/masking-views.component.ts:48]: Are these elements needed (accessMode, availabilityZones)?

File: src/app/business/delfin/masking-views/masking-views.component.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    getAllMaskingViews(storageId){
+        this.ds.getAllMaskingViews(storageId).subscribe((res)=>{
+            let allViews = res.json().masking_views
+            
+            
+            allViews.forEach(view => {
+                view['storageHostInitiatorsCount'] = 0;
+                view['volumesCount'] = 0;
```
Comment (sushanthakumar) [inline src/app/business/delfin/masking-views/masking-views.component.ts:76]: Is this displayed somewhere?

File: src/app/business/delfin/masking-views/masking-views.component.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            case 'storageHost': this.detailsLabels.heading = "Storage Host Details";
+              
+              break;
+            case 'volumesList': this.detailsLabels.heading = "Volumes List";
+              
+              break;
+            case 'portsList': this.detailsLabels.heading = "Ports List";
+              
+              break;
+            case 'initiatorsList': this.detailsLabels.heading = "Initiators List";
```
Comment (sushanthakumar) [inline src/app/business/delfin/masking-views/masking-views.component.ts:126]: Prefix storage host for initiator

File: src/app/business/delfin/masking-views/masking-views.component.ts (hunk 3 above)
Comment (anvithks) [reply src/app/business/delfin/masking-views/masking-views.component.ts:48]: Not needed but these are labels used for display of Hosts from our On premise service. 
These can be removed.

File: src/app/business/delfin/masking-views/masking-views.component.ts (hunk 4 above)
Comment (anvithks) [reply src/app/business/delfin/masking-views/masking-views.component.ts:76]: Not at the moment. The count was maintained earlier to show in the column.

**Comment by kumarashit** [comment]
Few comments:
a) Rather than 'Ports list' can we simply say 'Ports' or 'Port List'? same for Volumes List
b) When showing Storage Host details, Showing Storage ID and Native_storage_ID may be confusing. 
c) Storage Host detail should contain the Initiator

**Comment by anvithks** [comment]
> Few comments:
> a) Rather than 'Ports list' can we simply say 'Ports' or 'Port List'? same for Volumes List

Done.

> b) When showing Storage Host details, Showing Storage ID and Native_storage_ID may be confusing.

Yes. But they are two different fields and they have to be shown. This was done after discussion with @NajmudheenCT @sushanthakumar previously.

> c) Storage Host detail should contain the Initiator

Will make the change.

**Review by NajmudheenCT — COMMENTED** [review]
Some suggestions 
1. Initiators should be displayed in masking view table only if it has direct relation with initiator, otherwise show initiator under host view
2. Port lists -> ports
3. Storage host details -> change label ' native storage host ID' to 'native host id', since we are already in the context of storage, no need to append storage in the view

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Comment by anvithks** [comment]
> Some suggestions
> 
> 1. Initiators should be displayed in masking view table only if it has direct relation with initiator, otherwise show initiator under host view

Will check and change accordingly, if needed.

> 2. Port lists -> ports

Already changed.

> 3. Storage host details -> change label ' native storage host ID' to 'native host id', since we are already in the context of storage, no need to append storage in the view

Will change.

**Comment by anvithks** [comment]
@kumarashit @NajmudheenCT @sushanthakumar 
As per the comments and the discussions I have made the changes.
here are the updated screenshots.
**Storage host Details**
image
image

**ports**
image

**Volumes**
image

**Comment by NajmudheenCT** [comment]
latest masking view tab screenshot please..

**Comment by anvithks** [comment]
@NajmudheenCT here you go
image

**Review by NajmudheenCT — APPROVED** [review]
LGTM

**Pull request merged**